src/uucore/src/lib/features/checksum.rs

```diff
@@ -1214,6 +1285,75 @@ pub fn calculate_blake2b_length(length: usize) -> UResult<Option<usize>> {
     }
 }
 
+/// Validate BLAKE2b length with Fluent error messages
+/// This function is used by utilities that need localized error messages
+pub fn validate_blake2b_length(length_str: &str, utility_name: &str) -> UResult<Option<usize>> {
+    // First try to parse as u128 to handle very large numbers
+    match length_str.parse::<u128>() {
+        Ok(length_u128) => {
+            if length_u128 > usize::MAX as u128 {
+                // For very large numbers, always show the max length error
+                // Use the original string to avoid precision issues
+                let error_key = format!("{}-error-invalid-length", utility_name);
+                let max_key = format!("{}-error-max-digest-length", utility_name);
+                show_error!(
+                    "{}",
+                    translate!(&error_key, "length" => DisplayQuotable::quote(length_str))
+                );
+                return Err(io::Error::new(
+                    io::ErrorKind::InvalidInput,
+                    translate!(&max_key, "algorithm" => "BLAKE2b", "max_bits" => 512),
+                )
+                .into());
+            }
+            let length = length_u128 as usize;
+            match length {
+                0 => Ok(None),
+                n if n > 512 => {
+                    let error_key = format!("{}-error-invalid-length", utility_name);
+                    let max_key = format!("{}-error-max-digest-length", utility_name);
+                    show_error!(
+                        "{}",
+                        translate!(&error_key, "length" => DisplayQuotable::quote(length_str))
+                    );
+                    Err(io::Error::new(
+                        io::ErrorKind::InvalidInput,
+                        translate!(&max_key, "algorithm" => "BLAKE2b", "max_bits" => 512),
+                    )
+                    .into())
```

Comment:
This part is almost a duplicate of the ones above, there might be a way to factorize it with closures ?